From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 1 Aug 2017 22:18:41 +0200
Subject: [PATCH] WIP: Bring in baremetal support

- start by copying reclass/classes/cluster/virtual-mcp-ocata-ovs as
  classes/cluster/baremetal-mcp-ocata-ovs;
- add new state (maas) that will handle MaaS configuration;
- add explicit `saltutil.sync_all` call to copy Salt modules to
  mas01 node;

TODO:
- salt_contorl_xenial_image is still harcoded to x86_64 image from
  apt-mk.mirantis.com;
- better seperation between cfg01 and mas01 configuration;

Change-Id: Ie8dd584b140991d2bd992acdfe47f5644bf51409
Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 ci/deploy.sh                                       |   4 +
 mcp/config/scenario/baremetal/defaults.yaml        |   6 +
 .../baremetal/os-nosdn-nofeature-noha.yaml         |  14 ++
 mcp/config/states/maas                             |  11 ++
 .../haproxy_openstack_api.yml                      | 166 +++++++++++++++++++++
 .../cluster/baremetal-mcp-ocata-ovs/.gitkeep       |   0
 .../baremetal-mcp-ocata-ovs/infra/config.yml       |  75 ++++++++++
 .../cluster/baremetal-mcp-ocata-ovs/infra/init.yml |  19 +++
 .../cluster/baremetal-mcp-ocata-ovs/infra/maas.yml |  27 ++++
 .../cluster/baremetal-mcp-ocata-ovs/init.yml       |  24 +++
 .../baremetal-mcp-ocata-ovs/openstack/compute.yml  |  81 ++++++++++
 .../baremetal-mcp-ocata-ovs/openstack/control.yml  |  98 ++++++++++++
 .../baremetal-mcp-ocata-ovs/openstack/gateway.yml  |  81 ++++++++++
 .../baremetal-mcp-ocata-ovs/openstack/init.yml     | 123 +++++++++++++++
 .../nodes/cfg01.baremetal-mcp-ocata-ovs.local.yml  |  10 ++
 mcp/scripts/user-data.template                     |   4 +
 16 files changed, 743 insertions(+)
 create mode 100644 mcp/config/scenario/baremetal/defaults.yaml
 create mode 100644 mcp/config/scenario/baremetal/os-nosdn-nofeature-noha.yaml
 create mode 100755 mcp/config/states/maas
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-common/haproxy_openstack_api.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/.gitkeep
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/config.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/init.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/maas.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/init.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/compute.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/control.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/gateway.yml
 create mode 100644 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/init.yml
 create mode 100644 mcp/reclass/nodes/cfg01.baremetal-mcp-ocata-ovs.local.yml

diff --git a/ci/deploy.sh b/ci/deploy.sh
index 491b955..ded5ccf 100755
--- a/ci/deploy.sh
+++ b/ci/deploy.sh
@@ -135,6 +135,10 @@ URI_REGEXP='(file|https?|ftp)://.*'

 export SSH_KEY=${SSH_KEY:-mcp.rsa}
 export SALT_MASTER=${SALT_MASTER_IP:-192.168.10.100}
+# FIXME: Rework this after networking setup is settled for baremetal
+export MAAS_IP=${MAAS_IP:-192.168.10.3}
+export MAAS_PXE_IFACE=${MAAS_PXE_IFACE:-ens3}
+export PXE_GW=${PXE_GW:-192.168.10.1}
 export SSH_OPTS="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${SSH_KEY}"

 # Variables below are disabled for now, to be re-introduced or removed later
diff --git a/mcp/config/scenario/baremetal/defaults.yaml b/mcp/config/scenario/baremetal/defaults.yaml
new file mode 100644
index 0000000..b841e88
--- /dev/null
+++ b/mcp/config/scenario/baremetal/defaults.yaml
@@ -0,0 +1,6 @@
+base_image: https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img
+virtual:
+  default:
+    vcpus: 2
+    ram: 4096
+
diff --git a/mcp/config/scenario/baremetal/os-nosdn-nofeature-noha.yaml b/mcp/config/scenario/baremetal/os-nosdn-nofeature-noha.yaml
new file mode 100644
index 0000000..10ad159
--- /dev/null
+++ b/mcp/config/scenario/baremetal/os-nosdn-nofeature-noha.yaml
@@ -0,0 +1,14 @@
+cluster:
+  domain: baremetal-mcp-ocata-ovs.local
+  states:
+    - maas
+    - openstack
+    - neutron_compute
+    - networks
+virtual:
+  nodes:
+    - cfg01
+    - mas01
+  mas01:
+    vcpus: 4
+    ram: 16384
diff --git a/mcp/config/states/maas b/mcp/config/states/maas
new file mode 100755
index 0000000..8ed2f4c
--- /dev/null
+++ b/mcp/config/states/maas
@@ -0,0 +1,11 @@
+salt 'mas01*' cmd.run "add-apt-repository ppa:maas/stable"
+
+salt 'mas01*' saltutil.sync_all
+
+salt 'mas01*' state.apply linux,salt,openssh,ntp
+salt 'mas01*' state.apply linux.network.interface
+salt 'mas01*' state.apply maas.cluster,maas.region
+
+salt 'mas01*' pillar.item\
+  maas:region:admin:username \
+  maas:region:admin:password
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-common/haproxy_openstack_api.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-common/haproxy_openstack_api.yml
new file mode 100644
index 0000000..e63e9d5
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-common/haproxy_openstack_api.yml
@@ -0,0 +1,166 @@
+parameters:
+  _param:
+    haproxy_check: check inter 15s fastinter 2s downinter 4s rise 3 fall 3
+  haproxy:
+    proxy:
+      listen:
+        cinder_api:
+          type: openstack-service
+          service_name: cinder
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8776
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8776
+            params: ${_param:haproxy_check}
+        glance_api:
+          type: openstack-service
+          service_name: glance
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 9292
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 9292
+            params: ${_param:haproxy_check}
+        glance_registry_api:
+          type: general-service
+          service_name: glance
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 9191
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 9191
+            params: ${_param:haproxy_check}
+        glare:
+          type: general-service
+          service_name: glare
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 9494
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 9494
+            params: ${_param:haproxy_check}
+        heat_cloudwatch_api:
+          type: openstack-service
+          service_name: heat
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8003
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8003
+            params: ${_param:haproxy_check}
+        heat_api:
+          type: openstack-service
+          service_name: heat
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8004
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8004
+            params: ${_param:haproxy_check}
+        heat_cfn_api:
+          type: openstack-service
+          service_name: heat
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8000
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8000
+            params: ${_param:haproxy_check}
+        keystone_public_api:
+          type: openstack-service
+          service_name: keystone
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 5000
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 5000
+            params: ${_param:haproxy_check}
+        keystone_admin_api:
+          type: openstack-service
+          service_name: keystone
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 35357
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 35357
+            params: ${_param:haproxy_check}
+        neutron_api:
+          type: openstack-service
+          service_name: neutron
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 9696
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 9696
+            params: ${_param:haproxy_check}
+        nova_placement_api:
+          mode: http
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8778
+          options:
+          - httpclose
+          - httplog
+          health-check:
+            http:
+              options:
+              - expect status 401
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8778
+            params: ${_param:haproxy_check}
+        nova_ec2_api:
+          type: general-service
+          service_name: nova
+          check: false
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8773
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8773
+            params: ${_param:haproxy_check}
+        nova_api:
+          type: openstack-service
+          service_name: nova
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8774
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8774
+            params: ${_param:haproxy_check}
+        nova_metadata_api:
+          type: openstack-service
+          binds:
+          - address: ${_param:cluster_vip_address}
+            port: 8775
+          servers:
+          - name: ctl01
+            host: ${_param:cluster_node01_address}
+            port: 8775
+            params: ${_param:haproxy_check}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/.gitkeep b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/.gitkeep
new file mode 100644
index 0000000..e69de29
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/config.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/config.yml
new file mode 100644
index 0000000..a05c201
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/config.yml
@@ -0,0 +1,75 @@
+classes:
+- service.git.client
+- system.linux.system.single
+- system.linux.system.repo.mcp.salt
+- system.salt.master.api
+- system.salt.master.pkg
+- system.reclass.storage.salt
+- system.salt.minion.ca.salt_master
+- system.salt.minion.cert.proxy
+- system.mysql.client.single
+- system.reclass.storage.system.openstack_compute_multi
+- system.reclass.storage.system.openstack_gateway_single
+- system.reclass.storage.system.infra_maas_single
+- system.salt.control.cluster.infra_maas_single
+- cluster.baremetal-mcp-ocata-ovs
+parameters:
+  _param:
+    infra_maas_node01_hostname: mas01
+    infra_kvm_node01_hostname: kvm01
+    salt_control_xenial_image: http://apt-mk.mirantis.com/images/ubuntu-16-04-x64-mcp1.1.qcow2
+    openstack_control_node01_hostname: ctl01
+    reclass_data_repository: local
+    salt_master_environment_repository: "https://github.com/tcpcloud"
+    salt_master_environment_revision: master
+    reclass_config_master: 192.168.10.100
+    single_address: 172.16.10.100
+    salt_master_host: 127.0.0.1
+    salt_master_base_environment: prd
+    salt_minion_ca_host: ${linux:network:fqdn}
+    salt_api_password_hash: "$6$sGnRlxGf$al5jMCetLP.vfI/fTl3Z0N7Za1aeiexL487jAtyRABVfT3NlwZxQGVhO7S1N8OwS/34VHYwZQA8lkXwKMN/GS1"
+  linux:
+    network:
+      interface:
+        ens4:
+          enabled: true
+          type: eth
+          proto: static
+          address: ${_param:single_address}
+          netmask: 255.255.255.0
+  reclass:
+    storage:
+      data_source:
+        engine: local
+      node:
+        infra_maas_node01:
+          name: ${_param:infra_maas_node01_hostname}
+          domain: ${_param:cluster_domain}
+          classes:
+          - cluster.${_param:cluster_name}.infra.maas
+          params:
+            salt_master_host: ${_param:reclass_config_master}
+            single_address: ${_param:infra_maas_node01_deploy_address}
+        openstack_control_node01:
+          name: ${_param:openstack_control_node01_hostname}
+          domain: ${_param:cluster_domain}
+          classes:
+          - cluster.${_param:cluster_name}.openstack.control
+          params:
+            linux_system_codename: xenial
+            salt_master_host: ${_param:reclass_config_master}
+            single_address: ${_param:openstack_control_node01_address}
+        openstack_compute_node01:
+          params:
+            single_address: 172.16.10.105
+            tenant_address: 10.1.0.105
+            external_address: 10.16.0.105
+        openstack_compute_node02:
+          params:
+            single_address: 172.16.10.106
+            tenant_address: 10.1.0.106
+            external_address: 10.16.0.106
+        openstack_gateway_node01:
+          params:
+            tenant_address: 10.1.0.110
+            external_address: 10.16.0.110
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/init.yml
new file mode 100644
index 0000000..90fc9cf
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/init.yml
@@ -0,0 +1,19 @@
+parameters:
+  linux:
+    network:
+      host:
+        cfg01:
+          address: ${_param:infra_config_address}
+          names:
+          - cfg01
+          - cfg01.${_param:cluster_domain}
+        cfg:
+          address: ${_param:infra_config_address}
+          names:
+          - cfg
+          - cfg.${_param:cluster_domain}
+        mas01:
+          address: ${_param:infra_maas_node01_deploy_address}
+          names:
+          - mas01
+          - mas01.${_param:cluster_domain}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/maas.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/maas.yml
new file mode 100644
index 0000000..192ded5
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/infra/maas.yml
@@ -0,0 +1,27 @@
+classes:
+- system.linux.system.repo.saltstack.xenial
+- system.maas.region.single
+- cluster.baremetal-mcp-ocata-ovs
+parameters:
+  _param:
+    primary_interface: ens4
+    interface_mtu: 9000
+    linux_system_codename: xenial
+    maas_admin_username: opnfv
+    maas_admin_password: opnfv_secret
+    maas_db_password: opnfv_secret
+    dns_server01: 8.8.8.8
+  maas:
+    region:
+      salt_master_ip: 192.168.10.100
+  linux:
+    network:
+      interface:
+        primary_interface:
+          enabled: true
+          name: ${_param:primary_interface}
+          mtu: ${_param:interface_mtu}
+          proto: static
+          address: ${_param:single_address}
+          netmask: 255.255.255.0
+          type: eth
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/init.yml
new file mode 100644
index 0000000..3553f7c
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/init.yml
@@ -0,0 +1,24 @@
+classes:
+- system.linux.system.single
+- cluster.baremetal-mcp-ocata-ovs.infra
+- cluster.baremetal-mcp-ocata-ovs.openstack
+
+parameters:
+  _param:
+    cluster_domain: baremetal-mcp-ocata-ovs.local
+    cluster_name: baremetal-mcp-ocata-ovs
+    # infra service addresses
+    infra_config_address: 172.16.10.100
+    infra_maas_node01_deploy_address: 172.16.10.201
+    # openstack service addresses
+    openstack_control_address: 172.16.10.101
+    openstack_control_node01_address: 172.16.10.101
+    openstack_control_node02_address: 172.16.10.102
+    openstack_control_node03_address: 172.16.10.103
+    openstack_database_address: ${_param:openstack_control_address}
+    openstack_message_queue_address: ${_param:openstack_control_address}
+    openstack_message_queue_node01_address: ${_param:openstack_control_node01_address}
+    openstack_message_queue_node02_address: ${_param:openstack_control_node02_address}
+    openstack_message_queue_node03_address: ${_param:openstack_control_node03_address}
+    openstack_gateway_address: 172.16.10.110
+    control_address: ${_param:openstack_control_address}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/compute.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/compute.yml
new file mode 100644
index 0000000..d9c2db7
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/compute.yml
@@ -0,0 +1,81 @@
+classes:
+- system.linux.system.repo.mcp.openstack
+- system.linux.system.repo.mcp.extra
+- system.linux.system.repo.saltstack.xenial
+- system.linux.storage.loopback
+- system.nova.compute.single
+- service.neutron.compute.single
+- service.cinder.volume.single
+- system.cinder.volume.backend.lvm
+- cluster.baremetal-mcp-ocata-ovs
+parameters:
+  _param:
+    primary_interface: ens4
+    tenant_interface: ens5
+    external_interface: ens6
+    interface_mtu: 9000
+    linux_system_codename: xenial
+    loopback_device_size: 10
+  nova:
+    compute:
+      vncproxy_url: http://${_param:cluster_vip_address}:6080
+      network:
+        region: ${_param:openstack_region}
+        user: neutron
+        tenant: service
+        password: ${_param:keystone_neutron_password}
+  neutron:
+    compute:
+      agent_mode: ${_param:neutron_compute_agent_mode}
+      message_queue:
+        host: ${_param:openstack_control_address}
+      metadata:
+        host: ${_param:openstack_control_address}
+  cinder:
+    volume:
+      database:
+        host: ${_param:cluster_local_address}
+      identity:
+        host: ${_param:cluster_local_address}
+      glance:
+        host: ${_param:cluster_local_address}
+      message_queue:
+        host: ${_param:cluster_local_address}
+  linux:
+    network:
+      bridge: openvswitch
+      interface:
+        dhcp_int:
+          enabled: true
+          name: ens3
+          proto: dhcp
+          type: eth
+          mtu: ${_param:interface_mtu}
+        primary_interface:
+          enabled: true
+          name: ${_param:primary_interface}
+          mtu: ${_param:interface_mtu}
+          proto: manual
+          type: eth
+        tenant_interface:
+          enabled: true
+          name: ${_param:tenant_interface}
+          mtu: ${_param:interface_mtu}
+          proto: manual
+          type: eth
+        br-mgmt:
+          enabled: true
+          type: bridge
+          proto: static
+          address: ${_param:single_address}
+          netmask: 255.255.255.0
+          use_interfaces:
+          - ${_param:primary_interface}
+        br-mesh:
+          enabled: true
+          type: bridge
+          proto: static
+          address: ${_param:tenant_address}
+          netmask: 255.255.255.0
+          use_interfaces:
+          - ${_param:tenant_interface}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/control.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/control.yml
new file mode 100644
index 0000000..3bd7238
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/control.yml
@@ -0,0 +1,98 @@
+classes:
+- system.linux.system.lowmem
+- system.linux.system.repo.mcp.openstack
+- system.linux.system.repo.mcp.extra
+- system.linux.system.repo.saltstack.xenial
+- system.memcached.server.single
+- system.rabbitmq.server.single
+- system.rabbitmq.server.vhost.openstack
+- system.keystone.server.wsgi
+- system.keystone.server.single
+- system.keystone.client.single
+- system.keystone.client.service.nova21
+- system.keystone.client.service.nova-placement
+- system.keystone.client.service.glare
+- system.keystone.client.service.cinder3
+- system.glance.control.single
+- system.nova.control.single
+- system.neutron.control.openvswitch.single
+- system.cinder.control.single
+- system.cinder.control.backend.lvm
+- system.heat.server.single
+- service.mysql.server.single
+- system.galera.server.database.cinder
+- system.galera.server.database.glance
+- system.galera.server.database.grafana
+- system.galera.server.database.heat
+- system.galera.server.database.keystone
+- system.galera.server.database.nova
+- system.horizon.server.single
+- service.haproxy.proxy.single
+- cluster.baremetal-mcp-ocata-common.haproxy_openstack_api
+- cluster.baremetal-mcp-ocata-ovs
+parameters:
+  _param:
+    linux_system_codename: xenial
+  linux:
+    system:
+      package:
+        python-msgpack:
+          version: latest
+    network:
+      interface:
+        ens4:
+          enabled: true
+          type: eth
+          proto: static
+          address: ${_param:single_address}
+          netmask: 255.255.255.0
+        ens6:
+          enabled: true
+          type: eth
+          proto: static
+          address: ${_param:cluster_public_host}
+          netmask: 255.255.255.0
+  keystone:
+    server:
+      admin_email: ${_param:admin_email}
+      pkgs:
+      - keystone
+      - python-keystone
+      - python-keystoneclient
+      - python-psycopg2
+      - python-mysqldb
+      - python-six
+      - python-memcache
+      - python-openstackclient
+      - gettext-base
+      - python-pycadf
+  glance:
+    server:
+      storage:
+        engine: file
+      images: []
+      workers: 1
+  nova:
+    controller:
+      networking: dvr
+      cpu_allocation: 54
+      metadata:
+        password: ${_param:metadata_password}
+      bind:
+        private_address: ${_param:cluster_local_address}
+        public_address: ${_param:cluster_vip_address}
+        novncproxy_port: 6080
+      vncproxy_url: http://${_param:cluster_vip_address}:6080
+      workers: 1
+  heat:
+    server:
+      bind:
+        api_cfn:
+          address: ${_param:single_address}
+        api_cloudwatch:
+          address: ${_param:single_address}
+  mysql:
+    server:
+      version: '5.7'
+      bind:
+        address: ${_param:cluster_local_address}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/gateway.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/gateway.yml
new file mode 100644
index 0000000..38ca78c
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/gateway.yml
@@ -0,0 +1,81 @@
+classes:
+- system.linux.system.repo.mcp.openstack
+- system.linux.system.repo.mcp.extra
+- system.linux.system.repo.saltstack.xenial
+- service.neutron.gateway.single
+- cluster.baremetal-mcp-ocata-ovs
+parameters:
+  _param:
+    primary_interface: ens4
+    tenant_interface: ens5
+    external_interface: ens6
+    interface_mtu: 9000
+    linux_system_codename: xenial
+  neutron:
+    gateway:
+      agent_mode: ${_param:neutron_gateway_agent_mode}
+  linux:
+    network:
+      bridge: openvswitch
+      interface:
+        dhcp_int:
+          enabled: true
+          name: ens3
+          proto: dhcp
+          type: eth
+          mtu: ${_param:interface_mtu}
+        primary_interface:
+          enabled: true
+          name: ${_param:primary_interface}
+          mtu: ${_param:interface_mtu}
+          proto: manual
+          type: eth
+        tenant_interface:
+          enabled: true
+          name: ${_param:tenant_interface}
+          mtu: ${_param:interface_mtu}
+          proto: manual
+          type: eth
+        external_interface:
+          enabled: true
+          name: ${_param:external_interface}
+          mtu: ${_param:interface_mtu}
+          proto: manual
+          type: eth
+        br-floating:
+          enabled: true
+          type: ovs_bridge
+          mtu: ${_param:interface_mtu}
+        br-mgmt:
+          enabled: true
+          type: bridge
+          proto: static
+          address: ${_param:single_address}
+          netmask: 255.255.255.0
+          mtu: ${_param:interface_mtu}
+          use_interfaces:
+          - ${_param:primary_interface}
+        br-mesh:
+          enabled: true
+          type: bridge
+          mtu: ${_param:interface_mtu}
+          proto: static
+          address: ${_param:tenant_address}
+          netmask: 255.255.255.0
+          use_interfaces:
+          - ${_param:tenant_interface}
+        float-to-ex:
+          enabled: true
+          type: ovs_port
+          mtu: ${_param:interface_mtu}
+          bridge: br-floating
+        br-ex:
+          enabled: true
+          type: bridge
+          mtu: ${_param:interface_mtu}
+          address: ${_param:external_address}
+          netmask: 255.255.255.0
+          use_interfaces:
+          - ${_param:external_interface}
+          use_ovs_ports:
+          - float-to-ex
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/init.yml
new file mode 100644
index 0000000..7079fd1
--- /dev/null
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs/openstack/init.yml
@@ -0,0 +1,123 @@
+parameters:
+  _param:
+    openstack_version: ocata
+    apt_mk_version: nightly
+    mcp_repo_version: 1.1
+    openstack_region: RegionOne
+    admin_email: root@localhost
+    cluster_public_protocol: http
+    cluster_public_host: 10.16.0.101
+    neutron_public_protocol: http
+    neutron_control_dvr: False
+    neutron_tenant_network_types: "flat,vxlan"
+    neutron_l3_ha: False
+    neutron_global_physnet_mtu: 1500
+    neutron_external_mtu: 1500
+    neutron_gateway_dvr: False
+    neutron_gateway_agent_mode: legacy
+    neutron_compute_dvr: False
+    neutron_compute_agent_mode: legacy
+    neutron_compute_external_access: False
+    galera_server_cluster_name: openstack_cluster
+    galera_server_maintenance_password: opnfv_secret
+    galera_server_admin_password: opnfv_secret
+    cluster_vip_address: ${_param:cluster_public_host}
+    cluster_local_address: ${_param:openstack_control_address}
+    cluster_node01_hostname: ctl01
+    cluster_node01_address: 172.16.10.101
+    cluster_node02_hostname: ctl02
+    cluster_node02_address: 172.16.10.102
+    cluster_node03_hostname: ctl03
+    cluster_node03_address: 172.16.10.103
+    rabbitmq_secret_key: opnfv_secret
+    rabbitmq_admin_password: opnfv_secret
+    rabbitmq_openstack_password: opnfv_secret
+    rabbitmq_cold_password: opnfv_secret
+    glance_version: ${_param:openstack_version}
+    glance_service_host: ${_param:cluster_local_address}
+    keystone_version: ${_param:openstack_version}
+    keystone_service_host: ${_param:cluster_local_address}
+    heat_version: ${_param:openstack_version}
+    heat_service_host: ${_param:cluster_local_address}
+    heat_domain_admin_password: opnfv_secret
+    ceilometer_version: ${_param:openstack_version}
+    ceilometer_service_host: 172.16.10.108
+    ceilometer_database_host: ${_param:cluster_local_address}
+    cinder_version: ${_param:openstack_version}
+    cinder_service_host: ${_param:cluster_local_address}
+    ceilometer_graphite_publisher_host: 172.16.10.107
+    ceilometer_graphite_publisher_port: 2013
+    nova_version: ${_param:openstack_version}
+    nova_service_host: ${_param:cluster_local_address}
+    nova_vncproxy_url: http://${_param:cluster_vip_address}:8060
+    neutron_version: ${_param:openstack_version}
+    neutron_service_host: ${_param:cluster_local_address}
+    metadata_password: password
+    mysql_admin_user: root
+    mysql_admin_password: opnfv_secret
+    mysql_cinder_password: opnfv_secret
+    mysql_ceilometer_password: opnfv_secret
+    mysql_glance_password: opnfv_secret
+    mysql_grafana_password: opnfv_secret
+    mysql_heat_password: opnfv_secret
+    mysql_keystone_password: opnfv_secret
+    mysql_neutron_password: opnfv_secret
+    mysql_nova_password: opnfv_secret
+    mysql_aodh_password: opnfv_secret
+    keystone_service_token: opnfv_secret
+    keystone_admin_password: opnfv_secret
+    keystone_ceilometer_password: opnfv_secret
+    keystone_cinder_password: opnfv_secret
+    keystone_glance_password: opnfv_secret
+    keystone_heat_password: opnfv_secret
+    keystone_keystone_password: opnfv_secret
+    keystone_neutron_password: opnfv_secret
+    keystone_nova_password: opnfv_secret
+    ceilometer_secret_key: opnfv_secret
+    metadata_password: opnfv_secret
+    horizon_version: ${_param:openstack_version}
+    horizon_secret_key: opaesee8Que2yahJoh9fo0eefo1Aeyo6ahyei8zeiboh3aeth5loth7ieNa5xi5e
+    horizon_identity_host: ${_param:cluster_vip_address}
+    horizon_identity_encryption: none
+    horizon_identity_version: 3
+    mongodb_server_replica_set: ceilometer
+    mongodb_ceilometer_password: cloudlab
+    mongodb_admin_password: cloudlab
+    mongodb_shared_key: eoTh1AwahlahqueingeejooLughah4tei9feing0eeVaephooDi2li1TaeV1ooth
+    aodh_version: ${_param:openstack_version}
+    keystone_aodh_password: opnfv_secret
+    aodh_service_host: 172.16.10.108
+  linux:
+    system:
+      kernel:
+        sysctl:
+          net.ipv4.tcp_congestion_control: yeah
+          net.ipv4.tcp_slow_start_after_idle: 0
+          net.ipv4.tcp_fin_timeout: 30
+    network:
+      host:
+        ctl:
+          address: ${_param:openstack_control_address}
+          names:
+          - ctl
+          - ctl.${_param:cluster_domain}
+        ctl01:
+          address: ${_param:openstack_control_node01_address}
+          names:
+          - ctl01
+          - ctl01.${_param:cluster_domain}
+        gtw01:
+          address: ${_param:openstack_gateway_address}
+          names:
+          - gtw01
+          - gtw01.${_param:cluster_domain}
+        cmp01:
+          address: 172.16.10.105
+          names:
+          - cmp01
+          - cmp01.${_param:cluster_domain}
+        cmp02:
+          address: 172.16.10.106
+          names:
+          - cmp02
+          - cmp02.${_param:cluster_domain}
diff --git a/mcp/reclass/nodes/cfg01.baremetal-mcp-ocata-ovs.local.yml b/mcp/reclass/nodes/cfg01.baremetal-mcp-ocata-ovs.local.yml
new file mode 100644
index 0000000..fd8b0a0
--- /dev/null
+++ b/mcp/reclass/nodes/cfg01.baremetal-mcp-ocata-ovs.local.yml
@@ -0,0 +1,10 @@
+classes:
+- cluster.baremetal-mcp-ocata-ovs.infra.config
+parameters:
+  _param:
+    linux_system_codename: xenial
+    reclass_data_revision: master
+  linux:
+    system:
+      name: cfg01
+      domain: baremetal-mcp-ocata-ovs.local
diff --git a/mcp/scripts/user-data.template b/mcp/scripts/user-data.template
index c696d35..ec3796a 100644
--- a/mcp/scripts/user-data.template
+++ b/mcp/scripts/user-data.template
@@ -1,4 +1,8 @@
 #!/bin/bash
+if [ "$(hostname)" = "mas01" ]; then
+  ip address add ${MAAS_IP}/24 dev ${MAAS_PXE_IFACE}
+  ip route add default via ${PXE_GW}
+fi
 if [ "$(uname -i)" = "aarch64" ]; then
   wget -O - http://linux.enea.com/saltstack/apt/ubuntu/16.04/arm64/latest/SALTSTACK-GPG-KEY.pub | sudo apt-key add -
   echo "deb http://linux.enea.com/saltstack/apt/ubuntu/16.04/arm64/latest xenial main" > /etc/apt/sources.list.d/salt.list
